projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -192,11 +197,22 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
             // handle scenario where last child row might not be in view
             // parent should scroll to child grid end
             const childContainer = this.grid.nativeElement.parentNode.parentNode;
-            const diff =
+            const diffBottom =
             childContainer.getBoundingClientRect().bottom - this.grid.rootGrid.nativeElement.getBoundingClientRect().bottom;
-            const endIsVisible = diff < 0;
+            const row = this.grid.getRowByIndex(rowIndex).element.nativeElement;
+            const gridTop = this._getMaxTop(this.grid);
+            const diffTop = row.getBoundingClientRect().bottom -
+            row.offsetHeight - gridTop;
+            const endIsVisible = diffBottom <= 0;
+            const topVisible = diffTop >= 0;
             if (!endIsVisible) {
-                this.scrollGrid(this.grid.parent, diff, () => super.onKeydownEnd(rowIndex));
+                this.scrollGrid(this.grid.parent, diffBottom, () => super.onKeydownEnd(rowIndex));
+            } else if (!topVisible) {
+            const scrGrid = this.grid.verticalScrollContainer.getVerticalScroll().scrollTop !== 0 ? this.grid :
+             this.getNextScrollable(this.grid).grid;
+            const topGrid = scrGrid.tbody.nativeElement.getBoundingClientRect().top >
+            this.grid.rootGrid.tbody.nativeElement.getBoundingClientRect().top ? scrGrid : this.grid.rootGrid;
+            this.scrollGrid(topGrid, diffTop, () => super.onKeydownEnd(rowIndex));
             } else {
                 super.onKeydownEnd(rowIndex);
```

Comment:
`onKeydownEnd` takes `isSummary` as well. I think it should be implemented for this method override too.